examples/x86_64/src/logger.rs

```diff
@@ -0,0 +1,45 @@
+//! Log implementation using the UART.
+
+use core::fmt::Write;
+use log::{LevelFilter, Log, Metadata, Record, SetLoggerError};
+use spin::mutex::SpinMutex;
+use uart_16550::SerialPort;
+
+static LOGGER: Logger = Logger {
+    uart: SpinMutex::new(None),
+};
+
+struct Logger {
+    uart: SpinMutex<Option<SerialPort>>,
+}
+
+/// Initialises UART logger.
+pub fn init(max_level: LevelFilter) -> Result<(), SetLoggerError> {
+    // Safe because BASE_ADDRESS is the base of the MMIO region for a UART and is mapped as device
+    // memory.
```

Comment:
There's no MMIO or memory mapping involved here is there? I guess you mean that the I/O port belongs to a 16550 UART.